addons/payment_paypal/controllers/main.py

```diff
@@ -24,52 +26,104 @@ def paypal_dpn(self, **data):
         The route also allows the GET method in case the user clicks on "go back to merchant site".
         """
         _logger.info("beginning DPN with post data:\n%s", pprint.pformat(data))
-        try:
-            self._validate_data_authenticity(**data)
-        except ValidationError:
-            pass  # The transaction has been moved to state 'error'. Redirect to /payment/status.
+        if not data:  # The customer has cancelled the payment
+            pass  # Redirect them to the status page to browse the draft transaction
         else:
-            if data:
-                request.env['payment.transaction'].sudo()._handle_feedback_data('paypal', data)
+            try:
+                notification_data = self._validate_pdt_data_authenticity(**data)
+            except ValidationError:
+                pass  # The data authenticity could not be verified, discard the notification
```

Comment:
MIght be a good idea to log the validation error here instead of silencing it? Even if just as info/warning.